test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
+      assert.strictEqual(c.xAlignment(), xAlignment, "x alignment has been set");
 
-  it("can't reuse component if it's been destroy()-ed", () => {
-    let c1 = new Plottable.Component();
-    c1.renderTo(svg);
-    c1.destroy();
+      let yAlignment = Plottable.Components.Alignment.BOTTOM;
+      assert.strictEqual(c.yAlignment(yAlignment), c, "returns calling object");
+      assert.strictEqual(c.yAlignment(), yAlignment, "y alignment has been set");
+      c.destroy();
+      svg.remove();
+    });
 
-    assert.throws(() => c1.renderTo(svg), "reuse");
-    svg.remove();
-  });
+    it("throws errors on bad alignments", () => {
+      let invalidAlignment = "foo";
+      assert.throws(() => c.xAlignment(invalidAlignment), Error, "Unsupported alignment");
+      assert.throws(() => c.yAlignment(invalidAlignment), Error, "Unsupported alignment");
+      c.destroy();
+      svg.remove();
+    });
 
-  it("redraw() works as expected", () => {
-    let cg = new Plottable.Components.Group();
-    let c = TestMethods.makeFixedSizeComponent(10, 10);
-    cg.append(c);
-    cg.renderTo(svg);
-    assert.strictEqual(cg.height(), 300, "height() is the entire available height");
-    assert.strictEqual(cg.width(), 400, "width() is the entire available width");
-    TestMethods.fixComponentSize(c, 50, 50);
-    c.redraw();
-    assert.strictEqual(cg.height(), 300, "height() after resizing is the entire available height");
-    assert.strictEqual(cg.width(), 400, "width() after resizing is the entire available width");
-    svg.remove();
-  });
+    describe("on fixed size components", () => {
+      let fixedWidth = 100;
+      let fixedHeight = 100;
 
-  it("component remains in own cell", () => {
-    let horizontalComponent = new Plottable.Component();
-    let verticalComponent = new Plottable.Component();
-    let placeHolder = new Plottable.Component();
-    let t = new Plottable.Components.Table().add(verticalComponent, 0, 0)
-                                 .add(new Plottable.Component(), 0, 1)
-                                 .add(placeHolder, 1, 0)
-                                 .add(horizontalComponent, 1, 1);
-    t.renderTo(svg);
-    horizontalComponent.xAlignment("center");
-    verticalComponent.yAlignment("bottom");
-
-    TestMethods.assertBBoxNonIntersection((<any> verticalComponent)._element.select(".bounding-box"),
-                              (<any> placeHolder)._element.select(".bounding-box"));
-    TestMethods.assertBBoxInclusion((<any> t)._boxContainer.select(".bounding-box"),
-                                    (<any> horizontalComponent)._element.select(".bounding-box"));
+      beforeEach(() => {
+        TestMethods.fixComponentSize(c, fixedWidth, fixedHeight);
+        c.anchor(svg);
+      });
 
-    svg.remove();
+      it("translates the origin in accordance with alignment", () => {
+        c.xAlignment("left").yAlignment("top");
+        c.computeLayout();
+        let componentOrigin = {x: 0, y: 0};
```

Comment:
I would prefer it is named something like `expectedOrigin`, but I am fine either way